src/application/application.cpp

```diff
@@ -242,7 +253,8 @@ void Application::Predict() {
     objective_fun_->Init(train_data_->metadata(), train_data_->num_data());
     boosting_->Init(&config_, train_data_.get(), objective_fun_.get(),
                     Common::ConstPtrInVectorWrapper<Metric>(train_metric_));
-    boosting_->RefitTree(pred_leaf);
+
+    boosting_->RefitTree(pred_leaf.data(), nrow, pred_leaf.size() / nrow);
```

Comment:
This could be simpler, I think

```suggestion
    boosting_->RefitTree(pred_leaf.data(), nrow, ncol);
```